geonode/base/api/permissions.py

```diff
@@ -153,6 +154,32 @@ def has_object_permission(self, request, view, obj):
         return IsOwnerOrAdmin.has_object_permission(self, request, view, obj)
 
 
+class IsManagerEditOrAdmin(permissions.BasePermission):
+    """
+    Object-level permission to only allow admin and managers to edit a group.
+    """
+    def has_permission(self, request, view):
+        user = request.user
+        if request.method in ['POST', 'DELETE']:
+            return user and (user.is_superuser or user.is_staff)
```

Comment:
The user I think is better to be moved in the IF condition and leave only the check about the superuser/staff for the return.